docs/src/content/docs/guides/components.mdx

```diff
@@ -195,3 +195,49 @@ The code above generates the following on the page:
 A list of all available icons is shown below with their associated names. Click an icon to copy the component code for it.
 
 <IconsList />
+
+### Code
+
+Starlight uses Expressive Code to render [Markdown and MDX code blocks](/guides/authoring-content/#code-blocks). Use the `<Code>` component to render syntax highlighted code when using a Markdown code block is not possible, for example, to render data coming from external sources like files, databases, or APIs.
+
+See the [Expressive Code “Code Component” docs](https://expressive-code.com/key-features/code-component/) for full details of the props `<Code>` supports.
+
+```mdx
+# src/content/docs/example.mdx
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('This could come from a file or CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+The code above generates the following on the page:
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('This could come from a file or CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ["file", "CMS"];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+
+#### Code transclusion
```

Comment:
```suggestion
#### Imported Code
```
I don't hate "transclusion"! (though, I also didn't know what it was!) ... but wonder whether something more funtional here like "Imported Code" might help, since that's in fact how it's talked about in the explanation.

If you're going to say, "you wouldn't *always* do this with code you import" then I would reply that the wording below can be shifted to talk about the process of importing code, and then the part below becomes an "and if you want it imported as a string..." description.

Reply:
Yup I’m happy to use a simpler and more common word here too. I’d seen a few people ask us about this and use “transclusion”, but VitePress’s docs also say [“Import Code Snippets”](https://vitepress.dev/guide/markdown#import-code-snippets) for their equivalent feature if we need more convincing.